File: prefix.java
public class Main {
    public static String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }

        String prefix = strs[0];

        for (int i = 1; i < strs.length; i++) {
            while (prefix.length() > 0 &&
                    (strs[i].length() < prefix.length() ||
                            !strs[i].substring(0, prefix.length()).equals(prefix))) {
                prefix = prefix.substring(0, prefix.length() - 1);
            }

            if (prefix.isEmpty()) {
                return "";
            }
        }

        return prefix;
    }

    public static void main(String[] args) {
        String[] strs = {"flower", "flow", "flight"};

        String result = longestCommonPrefix(strs);

        if (result.isEmpty()) {
            System.out.println("Prefix mavjud emas.");
        } else {
            System.out.println("Eng uzun umumiy prefix: " + result);
        }
    }
}

File: prefix.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Println("Massiv so'zlarini kiriting:")
	input, _ := reader.ReadString('\n')
	words := strings.Fields(input)

	if len(words) == 0 {
		fmt.Println("Hech qanday so'z kiritilmadi")
		return
	}

	result := longestCommonPrefix(words)

	if result == "" {
		fmt.Println("Kiritilgan so'zlar ichida umumiy prefix mavjud emas.")
	} else {
		fmt.Println("Eng uzun umumiy prefix:", result)
	}
}

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}

	prefix := strs[0]

	for _, str := range strs[1:] {
		for len(prefix) > 0 && (len(str) < len(prefix) || str[:len(prefix)] != prefix) {
			prefix = prefix[:len(prefix)-1]
		}
		if prefix == "" {
			return ""
		}
	}

	return prefix
}
